runtime/util/annhelp.c

```diff
@@ -139,3 +143,122 @@ skipAnnotationElement(J9ROMConstantPoolItem const * constantPool, U_8 const *dat
 	*pIndex = index;
 	return -1;
 }
+
+/**
+ * Check if a field contains the specified Runtime Visible annotation.
+ *
+ * @param clazz The class the field belongs to.
+ * @param cpIndex The constant pool index of the field.
+ * @param annotationName The name of the annotation to check for.
+ * @return TRUE if the annotation is found, FALSE otherwise.
+ */
+BOOLEAN
+fieldContainsRuntimeAnnotation(J9VMThread *currentThread, J9Class *clazz, UDATA cpIndex, J9UTF8 *annotationName)
+{
+	BOOLEAN annotationFound = FALSE;
+	J9ROMClass *romClass = NULL;
+	J9ROMConstantPoolItem *constantPool = NULL;
+	J9ROMFieldRef *romFieldRef = NULL;
+	J9ROMFieldShape *romFieldShape = NULL;
+	J9Class *declaringClass = NULL;
+	J9ROMNameAndSignature *nameAndSig = NULL;
+	J9UTF8 *name = NULL;
+	J9UTF8 *signature = NULL;
+	J9Class *definingClass = NULL;
+	J9ConstantPool *ramCP = (J9ConstantPool *)clazz->ramConstantPool;
+
+	Assert_VMUtil_true(NULL != clazz);
+	Assert_VMUtil_true(NULL != annotationName);
+
+	romClass = clazz->romClass;
+	constantPool = ramCP->romConstantPool;
+	romFieldRef = (J9ROMFieldRef *)&constantPool[cpIndex];
+
+	declaringClass = ((J9RAMClassRef *)&ramCP[romFieldRef->classRefCPIndex])->value;
```

Comment:
This is not the declaring class of the field, it's the class where we start the search. `fieldClass` might be better.

Reply:
Or 'declaredClass` perhaps.